uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-balance.report.component.spec.js

```diff
@@ -0,0 +1,213 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'mocks/data/absence-type-data',
+  'mocks/data/leave-balance-report-data',
+  'mocks/apis/absence-type-api-mock',
+  'mocks/apis/leave-balance-report-api-mock',
+  'leave-absences/shared/models/leave-balance-report.model',
+  'leave-absences/shared/components/leave-balance-report.component'
+], function (absenceTypeMock, reportMockData) {
+  describe('LeaveBalanceReport.component', function () {
+    var $componentController, $provide, $q, $rootScope, AbsenceType, ctrl, leaveBalanceReport, notification, Session;
+    var loggedInContactId = 101;
+    var defaultReportSize = 50;
+
+    beforeEach(module('leave-absences.mocks', 'leave-absences.models', 'leave-absences.components', function (_$provide_) {
+      $provide = _$provide_;
+    }));
+
+    beforeEach(inject(function (_AbsenceTypeAPIMock_, _LeaveBalanceReportAPIMock_) {
+      $provide.value('AbsenceTypeAPI', _AbsenceTypeAPIMock_);
+      $provide.value('LeaveBalanceReportAPI', _LeaveBalanceReportAPIMock_);
+    }));
+
+    beforeEach(inject(function (_$componentController_, _$q_, _$rootScope_, _AbsenceType_, _LeaveBalanceReport_, _Session_, _notification_) {
+      $componentController = _$componentController_;
+      $q = _$q_;
+      $rootScope = _$rootScope_;
+      AbsenceType = _AbsenceType_;
+      leaveBalanceReport = _LeaveBalanceReport_;
+      notification = _notification_;
+      Session = _Session_;
+
+      spyOn(AbsenceType, 'all').and.callThrough();
+      spyOn(leaveBalanceReport, 'all').and.callThrough();
+      spyOn(notification, 'error');
+      spyOn(_Session_, 'get').and.returnValue($q.resolve({ contact_id: loggedInContactId }));
+    }));
+
+    describe('when the component is initializing', function () {
+      beforeEach(function () {
+        setupController();
+      });
+
+      it('sets .absenceTypes to an empty array', function () {
+        expect(ctrl.absenceTypes).toEqual([]);
+      });
+
+      it('sets .loading.report to false', function () {
+        expect(ctrl.loading.report).toBe(false);
+      });
+
+      it('sets .report to an empty array', function () {
+        expect(ctrl.report).toEqual([]);
+      });
+
+      it('sets .reportCount to 0', function () {
+        expect(ctrl.reportCount).toBe(0);
+      });
+
+      describe('when loading absence types', function () {
+        beforeEach(function () {
+          setupControllerAndRunOnInit();
+        });
+
+        it('calls AbsenceType.all to get a list of absence types', function () {
+          expect(AbsenceType.all).toHaveBeenCalledWith({
+            options: { sort: 'title ASC' }
```

Comment:
i think that the sorting is something that deserves a mention in the title "it loads the absence types sorted by title"